### ISSUE TITLE
List of users looking for players

### ISSUE THREAD

**Issue by qreeves** [red-eclipse/base#19]
From: http://redeclipse.net/forum/viewtopic.php?f=9&t=76

Let's say you tell someone "Let's go on x server", but then you change your decision. It would be nice to have some kind of user list to the left hand of the servers list that you can send a private message to someone and talk about which server to go to.

**Comment by Evropi** [comment]
Hmm, I interpreted this more as a local friends list. How would a player add another player to their friends list?

**Comment by freezurbern** [comment]
Perhaps be able to get the player auth names of all connected players on all servers, and if the auth name appears in a list locally (/friends = "jack jill joe"), highlight/flash that name on the players-connected part of the server list?

**Comment by Evropi** [comment]
That's the thing @freezurbern, I think few will use it if it's a command-line based function.

Most games do this by making the scoreboard interactive and letting you do various actions (e.g. add as friend, mute, etc) on the names listed there, but I reckon the whole scoreboard would need to be rewritten for that. The server menu has a similar pickle; few of the players are listed there, and the active region changing in a list menu for sub-items is generally bad UX practice.

So there needs to be a user-friendly, graphical way to do it.

**Comment by bonifarz** [comment]
I guess the scoreboard is fine without mouse interaction.
i once added a buddy list mod to the servers menu, pretty close to what freezurbern suggested, but it was pretty useless.
If you want to see on which servers your friends are playing, I still think a search filter for the server list is useful, but that's another topic.

**Comment by qreeves** [comment]
I've put a few "wishlist" items like this up on the issue tracker so new contributors have items they can attempt (and I'll probably put up many more if it works). The original topic went into a discussion about a "lobby" where people who are in the RE client are displayed, showing whether they are in-game, or waiting, or looking for someone to play with.

The query reply from servers now includes a list of auth names for each client connected to the server (if you hover over a server, all players and their associated auth names are listed). This could probably be used to compare to a "friends list" var. You could probably just have a text entry menu somewhere which contains a list of players the client can add to, next to which it would show which friends are online and on which server. Let's say, a "friends menu". This could be modified so when called up when on a server, you can add friends from the clients currently on the server.

**Comment by bonifarz** [comment]
Right, so "lobby" and "friend list" are different feature request. Just found the old mod I had in mind (http://forum.freegamedev.net/viewtopic.php?f=73&t=4928), maybe I'll give it a try later.
